### PR TITLE
PIM-5589: Change import system for currencies

### PR DESCRIPTION
**Pull request by pierallard**
Replace old currency import by 1.4 imports.

**Definition Of Done (for Core Developer only)**

```
| Q                                 | A
| --------------------------------- | ---
| Added Specs                       | y
| Added Behats                      | n
| Changelog updated                 | TODO
| Review and 2 GTM                  | TODO
| Micro Demo to the PO (Story only) | not needed
| Migration script                  | n
| Tech Doc                          | n
```

### CODE REVIEW

File: src/Pim/Component/Catalog/Updater/CurrencyUpdater.php [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+
+        return $this;
+    }
+
+    /**
+     * @param CurrencyInterface $currency
+     * @param string            $field
+     * @param mixed             $data
+     *
+     * @throws \InvalidArgumentException
```
Comment (skeleton) [inline src/Pim/Component/Catalog/Updater/CurrencyUpdater.php]: wrong

File: src/Pim/Component/Catalog/Updater/CurrencyUpdater.php [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+     */
+    protected function setData($currency, $field, $data)
```
Comment (skeleton) [inline src/Pim/Component/Catalog/Updater/CurrencyUpdater.php]: typehint

File: src/Pim/Bundle/CatalogBundle/EventSubscriber/CurrencyDisablingSubscriber.php [hunk 3]
```diff
@@ -50,7 +52,9 @@ public function checkChannelLink(GenericEvent $event)
             return;
         }
 
-        if (!$object->isActivated() && 0 < $this->channelRepository->getChannelCountUsingCurrency($object)) {
+        if (null !== $object->getId() &&
```
Comment (nidup) [inline src/Pim/Bundle/CatalogBundle/EventSubscriber/CurrencyDisablingSubscriber.php]: ok only for existing currency

**Comment by nidup** [comment]
baby comments then GTM

File: src/Pim/Bundle/CatalogBundle/Resources/config/savers.yml [hunk 4]
```diff
@@ -6,7 +6,6 @@ parameters:
     pim_catalog.saver.group_options_resolver.class:        Pim\Bundle\CatalogBundle\Doctrine\Common\Saver\GroupSavingOptionsResolver
     pim_catalog.saver.group.class:                         Pim\Bundle\CatalogBundle\Doctrine\Common\Saver\GroupSaver
     pim_catalog.saver.family.class:                        Pim\Bundle\CatalogBundle\Doctrine\Common\Saver\FamilySaver
-    pim_catalog.saver.channel.class:                       Pim\Bundle\CatalogBundle\Doctrine\Common\Saver\ChannelSaver
```
Comment (pierallard) [inline src/Pim/Bundle/CatalogBundle/Resources/config/savers.yml]: It never exists :)

**Comment by ClementGautier** [comment]
GTM

**Pull request closed without merging**